migrator/clone/db_clone_manager_impl.go

```diff
@@ -82,8 +85,28 @@ func (d *dbCloneManagerImpl) GetCloneToMigrate() (string, string, string, error)
 		}
 
 		// If we need to migrate from rocks we need to continue processing and
-		// get the Rocks clones.  If we don't, there is no need to process Rocks
+		// get the Rocks clones.  If we don't, there is no need to process Rocks, but
+		// we will check to see if we can get rid of rocks
 		if !migrateFromRocks {
+			currentPostgresVersion := d.dbmPostgres.GetCurrentVersion()
+			// We no longer need to migrate from Rocks so we should check to see if we have upgraded
+			// to the point where we should remove it entirely
+			if rocksVersion == nil || rocksVersion.SeqNum == 0 {
+				log.Info("rocks is empty or 0, going to decommission")
+				// Obviously if the Rocks version is nil or empty we need not carry the Rocks DB forward.
+				d.decommissionRocks = true
+			} else if currentPostgresVersion != nil &&
+				(version.CompareVersions(currentPostgresVersion.MainVersion, rocksVersion.MainVersion) > 0 &&
+					version.CompareVersions(version.GetMainVersion(), currentPostgresVersion.MainVersion) > 0) ||
```

Comment:
So you want to make sure you are in upgrade here. If we have Postgres enabled, we are in upgrade and we are not upgrade from rocks then we decomission.

There are issues in line 101. We cannot strictly compare the seqnum to make sure we are in an upgrade. In cases there is no migration added in this version, we may have an upgrade with the same sequence number.  And also we may not reliable compare that we are upgrading from a lower version to a higher version. There are cases we cannot determine which version is "larger" than the other for example in engineering versions, check ROX_DONT_COMPARE_DEV_BUILDS.
Those cases will end up not matching this condition.

Reply:
@c-du The general idea is that I tried to follow the same conditions that the clone managers use to determine if there is a rollback or an upgrade.  Those conditions check that the versions are different OR that the sequence numbers are different.  The only thing I'm doing here is doing the extra check to make sure we are planning to go multiple steps since our last RocksDB.  Like the check below, only I'm looking in the opposite direction:

`if version.CompareVersions(currClone.GetVersion(), version.GetMainVersion()) > 0 || currClone.GetSeqNum() > migrations.CurrentDBVersionSeqNum() {
			// Force rollback
			return PreviousClone, d.getPath(d.cloneMap[PreviousClone].GetDirName()), nil
		}`
		
Also, does it really matter for engineering versions?  In those cases we would just not decommission until such time that those versions were obviously compare higher.  What should we do in those cases?  Is that why the current logic like I posted above uses sequence number OR version?